plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -0,0 +1,856 @@
+    # must be removed from the beginning of Resource ID property value)
+    resource_id = "<<RESOURCE_ID>>"
+    # the metric names to collect
+    # leave the array empty to use all metrics available to this resource
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    # metrics aggregation type value to collect
+    # can be 'Total', 'Count', 'Average', 'Minimum', 'Maximum'
+    # leave the array empty to collect all aggregation types values for each metric
+    aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+  # resource target #2 to collect metrics from
+  [[inputs.azure_monitor.resource_target]]
+    resource_id = "<<RESOURCE_ID>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+
+  # resource group target #1 to collect metrics from resources under it with resource type
+  [[inputs.azure_monitor.resource_group_target]]
+    # the resource group name
+    resource_group = "<<RESOURCE_GROUP_NAME>>"
+
+    # defines the resources to collect metrics from
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      # the resource type
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+    # defines the resources to collect metrics from
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+      
+  # resource group target #2 to collect metrics from resources under it with resource type
+  [[inputs.azure_monitor.resource_group_target]]
+    resource_group = "<<RESOURCE_GROUP_NAME>>"
+
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+  
+  # subscription target #1 to collect metrics from resources under it with resource type    
+  [[inputs.azure_monitor.subscription_target]]
+    resource_type = "<<RESOURCE_TYPE>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+  # subscription target #2 to collect metrics from resources under it with resource type    
+  [[inputs.azure_monitor.subscription_target]]
+    resource_type = "<<RESOURCE_TYPE>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregation = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+`
+
+func (am *AzureMonitor) Description() string {
+	return "Gather Azure resources metrics from Azure Monitor API"
+}
+
+func (am *AzureMonitor) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (am *AzureMonitor) Init() error {
+	err := am.checkConfigValidation()
+
+	if err != nil {
+		return err
+	}
+
+	err = am.getAccessToken()
+
+	if err != nil {
+		return err
+	}
```

Comment:
You should add some prefix to the error to easier find the spot if seeing an error message and you can shorten this to
```suggestion
	if err := am.getAccessToken(); err != nil {
		return fmt.Errorf("getting access token failed: %v", err)
	}
```

This has the advantage that `err` is a local variable with the scope of the `if` part...